euphoria-spark/src/main/java/cz/seznam/euphoria/spark/BatchJoinIterator.java

```diff
@@ -143,4 +165,57 @@ private void emitLeft() {
   private boolean sameKey(Tuple2<BatchJoinKey<K>, ?> a, Tuple2<BatchJoinKey<K>, ?> b) {
     return Objects.equals(a._1.getKey(), b._1.getKey());
   }
+
+  private void addElementToStats(Tuple2<BatchJoinKey<K>, Either<L, R>> element){
+
+    K key = element._1.getKey();
+
+    Tuple2<MutableLong, MutableLong> leftRightCounts = numOfElementsByKey
+        .computeIfAbsent(key, (k) -> new Tuple2<>(new MutableLong(), new MutableLong()));
+
+    Either<L, R> eitherSide = element._2;
+
+    if(eitherSide.isLeft()){
+      leftRightCounts._1.increment();
+    }else {
+      leftRightCounts._2.increment();
+    }
+
+    numOfEncounteredElements++;
+  }
+
+  private void logStats(){
+
+    if(!LOG.isInfoEnabled()){
```

Comment:
Absolutely minor comment - we should be consistent in code style - if is not a function and therefore is followed by space. It just "looks better" :-)